includes/class-wc-adapter.php

```diff
@@ -0,0 +1,682 @@
+			OrderInterface::SHIPPING_POSTCODE     => $order->get_shipping_postcode(),
+			OrderInterface::SHIPPING_PHONE        => $order->get_billing_phone(),
+			OrderInterface::SHIPPING_EMAIL        => $order->get_billing_email(),
+			OrderInterface::SHIPPING_FIRST_NAME   => $order->get_shipping_first_name(),
+			OrderInterface::SHIPPING_LAST_NAME    => $order->get_shipping_last_name(),
+			OrderInterface::CUSTOMER_ID           => (int) $order->get_customer_id(),
+			OrderInterface::CUSTOMER_IP           => $order->get_customer_ip_address(),
+			OrderInterface::PAYER_REFERENCE       => $payer_reference,
+			OrderInterface::ITEMS                 => apply_filters( 'swedbank_pay_order_items', $items, $order ),
+			OrderInterface::LANGUAGE              => $this->getConfiguration()[ ConfigurationInterface::LANGUAGE ],
+		);
+	}
+
+	/**
+	 * Get Risk Indicator of Order.
+	 *
+	 * @param mixed $order_id
+	 *
+	 * @return array
+	 */
+	public function getRiskIndicator( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		$result = array();
+
+		// Downloadable
+		if ( $order->has_downloadable_item() ) {
+			// For electronic delivery, the email address to which the merchandise was delivered
+			$result[ RiskIndicatorInterface::DELIVERY_EMAIL_ADDRESS ] = $order->get_billing_email();
+
+			// Electronic Delivery
+			$result[ RiskIndicatorInterface::DELIVERY_TIME_FRAME_INDICATOR ] = '01';
+
+			// Digital goods, includes online services, electronic giftcards and redemption codes
+			$result[ RiskIndicatorInterface::SHIP_INDICATOR ] = '05';
+		}
+
+		// Shippable
+		if ( $order->needs_processing() ) {
+			// Two-day or more shipping
+			$result['deliveryTimeFrameIndicator'] = '04';
+
+			// Compare billing and shipping addresses
+			$billing  = $order->get_address( 'billing' );
+			$shipping = $order->get_address( 'shipping' );
+			$diff     = array_diff( $billing, $shipping );
+			if ( 0 === count( $diff ) ) {
+				// Ship to cardholder's billing address
+				$result[ RiskIndicatorInterface::SHIP_INDICATOR ] = '01';
+			} else {
+				// Ship to address that is different than cardholder's billing address
+				$result[ RiskIndicatorInterface::SHIP_INDICATOR ] = '03';
+			}
+		}
+
+		// @todo Add features of WooThemes Order Delivery and Pre-Orders WooCommerce Extensions
+
+		return apply_filters( 'swedbank_pay_risk_indicator', $result, $order, $this );
+	}
+
+	/**
+	 * Get Payee Info of Order.
+	 *
+	 * @param mixed $order_id
+	 *
+	 * @return array
+	 */
+	public function getPayeeInfo( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		return array(
+			PayeeInfoInterface::ORDER_REFERENCE => $order->get_id(),
+		);
+	}
+
+	/**
+	 * Update Order Status.
+	 *
+	 * @param mixed $order_id
+	 * @param string $status
+	 * @param string|null $message
+	 * @param mixed|null $transaction_id
+	 */
+	public function updateOrderStatus( $order_id, $status, $message = null, $transaction_id = null ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method updateOrderStatus() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)